File: from-1401-to-1500/1441-build-an-array-with-stack-operations/solution.py
class Solution(object):
    def buildArray(self, target, n):
        """
        :type target: List[int]
        :type n: int
        :rtype: List[str]
        """
        result = []
        
        for nxt in range(1, n+1):  
            
            result.append("Push")
            
            if nxt not in target:
                result.append("Pop")
                
            if nxt == target[-1]:
                break
                
        return result
                

File: from-1401-to-1500/1441-build-an-array-with-stack-operations/solution.go
func buildArray(target []int, n int) []string {
    result := []string{}
    targetMap := make(map[int]bool)
    for _, tgt := range target {
        targetMap[tgt] = true
    }
    goal := target[len(target) - 1]

    for nxt := 1; nxt <= n; nxt++ {
        result = append(result, "Push")
        if !targetMap[nxt] {
            result = append(result, "Pop")
        }
        if nxt == goal {
            break
        }
    }
    return result
}